### ISSUE TITLE
llegalStateException: Must call init() before using the store

### ISSUE THREAD

**Issue by gerostam** [iamMehedi/Secured-Preference-Store#40]
Hello,

I am using the library and works fine in all the devices that I tested so far. The problem is that I receive crashes in production for some older devices (i.e. Samsung Note 3 with Android 5):

llegalStateException: Must call init() before using the store in SecuredPreferenceStore.setRecoveryHandler

, but I call init as it is stated in the documentation (cannot replicate it on devices that I have available).

Please provide a fix and let me know!

**Comment by iamMehedi** [comment]
Most probably a memory issues, will check!

**Comment by lucianommoura** [comment]
> Most probably a memory issues, will check!

I dont think so. I have the same error at my application and all users have a lot of memory free.

**Comment by edgar-zigis** [comment]
We have a big user base, so I can confirm, that 95% of the crashes occur on Android 5 and earlier and the other 5% are occuring on Android 6.0-6.0.1.

**Comment by gerostam** [comment]
What I did to solve this (this is not the best solution as the library should not crash) is:

SecuredPreferenceStore securedPreferenceStore;
try {
      securedPreferenceStore = SecuredPreferenceStore.getSharedInstance();
} catch (Exception e) {
       LOGGER.error(e.toString(), e);
       generateSecuredSharedPreferences(context);
       securedPreferenceStore = SecuredPreferenceStore.getSharedInstance();
}


, when I try to use the the SecuredPreferenceStore Singleton
